### PR TITLE
Update installation instructions regarding CUDA

### PR DESCRIPTION
**Pull request by RaulPPelaez**
cc @peastman 
Follow up from #249

### CODE REVIEW

**Comment by peastman** [comment]
I would rather have it install CUDA 12 versions of everything.  I tried

```
CONDA_OVERRIDE_CUDA="12" mamba env create -f environment.yml
```

but it still installs 11.8.  I'm not sure where it's getting that from.  Presumably it must be required by one of the dependencies listed in environment.yaml?  The computer has a very recent driver and CUDA 12.2.

**Comment by RaulPPelaez** [comment]
Trying to get to the bottom of this here: https://github.com/conda-forge/cuda-version-feedstock/issues/14

File: docs/source/installation.rst
Comment (sef43) [inline docs/source/installation.rst:1]: the rst markup formatting is not quite correct on this file. At least looking at the version rendered by github: https://github.com/RaulPPelaez/torchmd-net/blob/docs_update/docs/source/installation.rst

File: README.md
Comment (sef43) [inline README.md:1]: can we just link to the installation.rst file instead of duplicating error prone instructions?

**Review by sef43 — CHANGES_REQUESTED** [review]

File: README.md
Comment (RaulPPelaez) [reply README.md:1]: I agree!

**Comment by RaulPPelaez** [comment]
I have not gotten a response yet from mamba guys. Perhaps the way to go here is not recommend mamba for the "from source" installation instructions.

**Comment by sef43** [comment]
Alternatively we just point people to the CUDA docs and tell them to install nvcc + extra from the official NVIDIA channels?
https://docs.nvidia.com/cuda/cuda-installation-guide-linux/index.html#conda-installation
Making a full nvcc compiler setup from conda is good for CI and deployment but not ideal for a user who probably already has a working nvcc on their system

**Comment by RaulPPelaez** [comment]
Well, official instructions include installing using conda from the nvidia channel. The other ones normally require root access, which I do not like given how we manage to get all the rest of the dependencies without it.
It is true though that in the end the target machine requires the correct driver anyway, so might as well get CUDA in the process. Let me give it a try and see if I can put everything in there...

**Comment by sef43** [comment]
The latest conda install commands work for me for cuda 12

**Comment by RaulPPelaez** [comment]
I am going to merge this.

**Comment by peastman** [comment]
I'm trying to follow the instructions in this PR.  Conda has been running for about an hour trying to solve the environment for the `conda env create` command. :(

**Comment by sef43** [comment]
Note that new and up to date conda will actually use libmamba solver backend. So for me the conda install was just as fast as the mamba one. But then I don't know if the original mamba bug in this issue still occurs.

https://conda.org/blog/2023-07-05-conda-libmamba-solver-rollout/

**Comment by peastman** [comment]
So it might be faster but have the same problem as mamba...  How useful!

It did finally succeed, and it installed CUDA 12.0.  It's now working on the command to install the rest of the CUDA libraries.

**Comment by peastman** [comment]
Everything eventually succeeded and I now have a working install.  It just took patience.  Thanks!

**Pull request merged**